### PR TITLE
Created functionality for tours duplication along with its steps

### PR DESCRIPTION
**Pull request by khu5h1**
### Summary of Changes
Created functionality for tours duplication along with its steps

### CODE REVIEW

File: administrator/components/com_guidedtours/src/Controller/ToursController.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 namespace Joomla\Component\Guidedtours\Administrator\Controller;
 
 use Joomla\CMS\MVC\Controller\AdminController;
+use Joomla\CMS\Language\Text;
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Controller/ToursController.php]:
```suggestion
use Joomla\CMS\Language\Text;
use Joomla\CMS\MVC\Controller\AdminController;
```

alpha sort

File: administrator/components/com_guidedtours/src/Controller/ToursController.php [hunk 2]
```diff
@@ -39,4 +40,21 @@ public function getModel($name = 'Tour', $prefix = 'Administrator', $config = ar
...
         return parent::getModel($name, $prefix, $config);
     }
+    public function duplicate()
+    {
+        $this->checkToken();
+        $pks = (array) $this->input->post->get('cid', array(), 'int');
+        $pks = array_filter($pks);
+        try {
+            if (empty($pks)) {
+                throw new \Exception(Text::_('COM_MODULES_ERROR_NO_MODULES_SELECTED'));
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Controller/ToursController.php]: copy paste error?

File: administrator/components/com_guidedtours/src/Model/TourModel.php [hunk 3]
```diff
@@ -16,6 +16,7 @@
 use Joomla\CMS\MVC\Model\AdminModel;
 use Joomla\CMS\Object\CMSObject;
 use Joomla\String\StringHelper;
+use Joomla\Database\ParameterType;
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Model/TourModel.php]:
```suggestion
use Joomla\Database\ParameterType;
use Joomla\String\StringHelper;
```

File: administrator/components/com_guidedtours/src/Model/TourModel.php [hunk 4]
```diff
@@ -308,4 +309,145 @@ public function getItem($pk = null)
...
+                $m = null;
+
+                if (preg_match('#\((\d+)\)$#', $table->title, $m)) {
+                    $table->title = preg_replace('#\(\d+\)$#', '(' . ($m[1] + 1) . ')', $table->title);
+                }
+
+                $data = $this->generateNewTitle(0, Text::_($table->title),Text::_($table->title));
+                $table->title = $data[0];
+
+                // Unpublish duplicate module
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Model/TourModel.php]: copy paste error?

File: administrator/language/en-GB/com_guidedtours.ini [hunk 5]
```diff
@@ -82,6 +82,8 @@ COM_GUIDEDTOURS_STEPS_LIST="Guided Tour: %s"
 COM_GUIDEDTOURS_TITLE="Guided Tours"
 COM_GUIDEDTOURS_TOUR_ID="ID"
 COM_GUIDEDTOURS_TOUR_TITLE="Title"
+COM_GUIDEDTOURS_TOURS_DUPLICATED="%d tours duplicated."
+COM_GUIDEDTOURS_TOURS_DUPLICATED_1="Tours duplicated."
```
Comment (brianteeman) [inline administrator/language/en-GB/com_guidedtours.ini]:
```suggestion
COM_GUIDEDTOURS_TOURS_DUPLICATED_1="Tour duplicated."
```

singular

File: administrator/components/com_guidedtours/src/Model/TourModel.php (hunk 4 above)
Comment (khu5h1) [reply administrator/components/com_guidedtours/src/Model/TourModel.php]: May you please let me know which error I need to change in this case?

Comment (brianteeman) [reply administrator/components/com_guidedtours/src/Model/TourModel.php]: it is just the comment that is wrong. I think you mean "Unpublish duplicate tour" ??

Comment (khu5h1) [reply administrator/components/com_guidedtours/src/Model/TourModel.php]: Yeah. Updating it. Thank you!

**Pull request merged**